compiler/rustc_codegen_llvm/src/llvm_util.rs

```diff
@@ -124,6 +125,35 @@ unsafe fn configure_llvm(sess: &Session) {
     llvm::LLVMRustSetLLVMOptions(llvm_args.len() as c_int, llvm_args.as_ptr());
 }
 
+fn save_invocation_args_for_debuginfo() {
+    fn os_to_cstr(os: &std::ffi::OsStr) -> Option<CString> {
+        let s = os.to_str()?;
+        CString::new(s).ok()
+    }
+
+    let full_args: Vec<CString> = std::env::args_os().map(|s| os_to_cstr(&s)).flatten().collect();
```

Comment:
should this also include the arguments parsed from `@argfiles`? having the path to the argfile doesn't seem very useful if you don't know the contents ... and in any case tools can pass arbitrary arguments through rustc_driver that weren't passed on the command line.

i'd suggest taking a look around `Session` to see if it stores the expanded arguments somewhere.

Reply:
Thanks for mentioning this. I did not now about that feature!
Yes I think replacing '@filepath' with the contents makes sense. I'll adpat my code.

Reply:
The expanded args were dropped after parsing. I added members to Session (and Config to pass it to Session) to store them. They are only needed as strings for adding to debuginfo, so I just used empty vecs in the tests code.

Reply:
thanks! can you also add a test to either `tests/debuginfo` or `tests/run-make` to confirm this debuginfo actually shows up in the generated binary?

Reply:
Will try. I only verified it manually so far.
It does seem a bit tricky though: This new debuginfo would only present in PDB files. I have not seen any kind of test that works with those.

Ideally, I'd do that with a PDB reader library, but that would require adding a new dependency or some prebuilt binary.
So I think I'd compile something and search within the PDB for the strings of that debuginfo. Even if I ignore the file format itself, checking for the presense of those strings might be enough for a test I hope.

Reply:
hmm, that seems better than nothing, but ideally we'd parse the format to make sure it's valid. maybe it makes sense to add a new test suite to compiletest so you can use a dependency there for the parsing? not sure

Reply:
I am having trouble using the existing tests. I don't know where the pdb file ends up (if its even created). I tried passing the -g flag in a run-make test.
I am strongly considering adding a new testsuite for PDBs, as you suggested. But might that be too big of a change for one PR?

Reply:
Ok my current state is that I"ll either try to get this running via run-make tests using llvm-pdbutil, or wait until https://github.com/rust-lang/rust/pull/113026 is merged and use a library then.

Discussion on Zulip: https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/Adding.20PDB.20tests